apis/go.mod

```diff
@@ -6,30 +6,57 @@ require (
 	github.com/google/uuid v1.3.0
 	github.com/metal3-io/baremetal-operator/pkg/hardwareutils v0.2.0
 	github.com/pkg/errors v0.9.1
-	github.com/stretchr/testify v1.8.0
-	k8s.io/api v0.26.1
-	k8s.io/apimachinery v0.26.1
-	sigs.k8s.io/controller-runtime v0.14.5
+	github.com/stretchr/testify v1.8.1
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
+	sigs.k8s.io/controller-runtime v0.15.0
 )
 
 require (
+	github.com/beorn7/perks v1.0.1 // indirect
```

Comment:
Ah, that's unfortunate :slightly_frowning_face:

Reply:
I wonder if the whole webhook code can be moved to another (maybe main) package. Or am I missing something about kubernetes (again)?

Reply:
I think it's here mainly because kubebuilder generates it this way and we didn't want to depart too radically from that because it makes it harder to add e.g. new webhooks.
I seem to recall that I did propose a way that it _could_ be done but it was complicated:

1. Stop automatically [registering](https://github.com/metal3-io/baremetal-operator/blob/main/apis/metal3.io/v1alpha1/baremetalhost_types.go#L1083) the GVK at startup.
2. Create a struct in the main module that embeds the API one, and register this instead.
3. Implement the Validate methods on this new struct type instead.
4. Consumers of the api module have to manually register the API's struct (we could provide a subpackage you could import to do it at automatically).

I guess we decided against, but either way it's definitely out of scope for this PR.